### ISSUE TITLE
getState micro or cluster

### ISSUE THREAD

**Issue by noeliaferruz** [Acellera/htmd#60]
Hi guys,

I wanted to print out around 1000 pdbs representative of the microstates. So I did:

```
mols=model.getStates(statetype='micro',numsamples=1)
for m in mols: ...
```
However this gives 3 states ( which is also the number of macros) instead of ~1000 which is the number of micros. The same with 'clusters'.

Suggestions?
Noelia

**Comment by j3mdamas** [comment]
Hola Noelia,

You have to do: `mols=model.getStates(states=range(model.micronum),statetype='micro',numsamples=1)`

It's a small bug, because the default `states` are set to `None`, and when that happens, the method sets it to `range(self.macronum)`. Let me know if it works, but it should. I'll fix it.

**Comment by noeliaferruz** [comment]
Yesss! Worked, thanks!

**Comment by j3mdamas** [comment]
:) Good. I'll fix it soon.

On Fri, Jun 10, 2016 at 8:56 PM, Noelia Ferruz <notifications@github.com>
wrote:

> Yesss! Worked, thanks!
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Acellera/htmd/issues/60#issuecomment-225266946>, or mute
> the thread
> <https://github.com/notifications/unsubscribe/AKTzQnwxZvsP0C75f_aXQ1m1pJmWCxV1ks5qKbNugaJpZM4IzMu2>
> .
>



--

**Comment by j3mdamas** [comment]
Fixed on multiscalelab/htmd@d76e2be973d51cb44ab18392805fbda93ed2a05f
